packages/library-base/src/commonMain/kotlin/io/realm/internal/SuspendableNotifier.kt

```diff
@@ -92,46 +93,58 @@ internal class SuspendableNotifier(
     }
 
     internal fun <T, C> registerObserver(thawableObservable: Thawable<Observable<T, C>>): Flow<C> {
-        return callbackFlow {
-            val token: AtomicRef<Cancellable> = kotlinx.atomicfu.atomic(NO_OP_NOTIFICATION_TOKEN)
-            withContext(dispatcher) {
-                ensureActive()
-                val liveRef: Observable<T, C> = thawableObservable.thaw(realm.realmReference)
-                    ?: error("Cannot listen for changes on a deleted reference")
-                val interopCallback: io.realm.internal.interop.Callback =
-                    object : io.realm.internal.interop.Callback {
-                        override fun onChange(change: NativePointer) {
-                            // FIXME How to make sure the Realm isn't closed when handling this?
+        var cancelCallback: () -> Unit = {}
 
-                            // FIXME The Realm should have been frozen in `realmChanged`, but this isn't supported yet.
-                            //  Instead we create the frozen version ourselves (which is correct, but pretty inefficient)
-                            //  We also send it to the owner Realm, so it can keep track of its lifecycle
-                            val frozenRealm = RealmReference(
-                                owner,
-                                RealmInterop.realm_freeze(realm.realmReference.dbPointer)
-                            )
-                            notifyRealmChanged(frozenRealm)
-                            // Notifications need to be delivered with the version they where created on, otherwise
-                            // the fine-grained notification data might be out of sync.
-                            liveRef.emitFrozenUpdate(
-                                frozenRealm,
-                                change,
-                                this@callbackFlow
-                            )?.let {
-                                checkResult(it)
+        return object :
+            Cancellable,
+            Flow<C> by callbackFlow({
+                cancelCallback = {
+                    cancel()
+                }
+                val token: AtomicRef<Cancellable> =
+                    kotlinx.atomicfu.atomic(NO_OP_NOTIFICATION_TOKEN)
+                withContext(dispatcher) {
+                    ensureActive()
+                    val liveRef: Observable<T, C> = thawableObservable.thaw(realm.realmReference)
+                        ?: error("Cannot listen for changes on a deleted reference")
+                    val interopCallback: io.realm.internal.interop.Callback =
+                        object : io.realm.internal.interop.Callback {
+                            override fun onChange(change: NativePointer) {
+                                // FIXME How to make sure the Realm isn't closed when handling this?
+
+                                // FIXME The Realm should have been frozen in `realmChanged`, but this isn't supported yet.
```

Comment:
I thought this was merged as part of the schema PR? https://github.com/realm/realm-kotlin/pull/539

Reply:
Although, I guess you just moved code around, so this should probably be addressed in another PR?

Reply:
You couldn't see the changes from #539 because they weren't merged in here yet.